File: 2283-check-if-number-has-equal-digit-count-and-digit-value/2283-check-if-number-has-equal-digit-count-and-digit-value.cpp
class Solution {
public:
    bool digitCount(string num) {
        unordered_map<int, int> hashMap;
        
        for(char n: num) {
            hashMap[n - '0']++;
        }
        
        for(int i = 0; i < num.length(); ++i) {
            if(hashMap[i] != (num[i] - '0'))
                return false;
        }
        return true;
    }
};

File: 2283-check-if-number-has-equal-digit-count-and-digit-value/2283-check-if-number-has-equal-digit-count-and-digit-value.py
class Solution:
    def digitCount(self, num: str) -> bool:
        hashMap = dict()
        
        for n in num:
            hashMap[int(n)] = hashMap.get(int(n), 0) + 1
        
        for i in range(len(num)):
            if hashMap.get(i, 0) != int(num[i]):
                return False
        return True